File: backend/Models/Application.cs
namespace JobTracker.Models
{
    public class Application
    {
        public int Id { get; set; }
        public required string CompanyName { get; set; }
        public required string JobTitle { get; set; }
        public required string Description { get; set; }
        public DateTimeOffset AppliedDate { get; set; }
        public DateTimeOffset LastUpdated { get; set; }
        public ApplicationStatus Status { get; set; }
    }

}

File: frontend/src/types/Application.ts
import type { ApplicationStatus } from "./ApplicationStatus";
import type { User } from "./User";

export interface Application{
    id: number,
    userId: number,
    user?: User,
    companyName: string,
    jobTitle: string,
    description: string,
    appliedDate: Date,
    lastUpdated: Date,
    status: ApplicationStatus
}